File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Cpp/src/LaptopDisplay.cpp
#include "../include/LaptopDisplay.h"

LaptopDisplay::LaptopDisplay(Subject* subject) : subject(subject) {
    subject->addObserver(this);
}

void LaptopDisplay::update(float temperature) {
    std::cout << "Laptop Display: Temperature updated to " << temperature << "°C\n";
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Observer/Java/LaptopDisplay.java
public class LaptopDisplay implements Observer {
    private Subject weatherStation;

    public LaptopDisplay(Subject weatherStation) {
        this.weatherStation = weatherStation;
        weatherStation.addObserver(this);
    }

    @Override
    public void update(float temperature) {
        System.out.println("Laptop Display: Temperature updated to " + temperature + "°C");
    }
}
